File: app/src/main/java/com/jakebarnby/filemanager/ui/adapters/OnSearchResultClicked.java
package com.jakebarnby.filemanager.ui.adapters;

import com.jakebarnby.filemanager.sources.models.SourceFile;
import com.jakebarnby.filemanager.util.TreeNode;

public interface OnSearchResultClicked {
    void navigateToFile(TreeNode<SourceFile> toOpen);
}

File: app/src/main/java/com/jakebarnby/filemanager/ui/adapters/OnSearchResultClicked.kt
package com.jakebarnby.filemanager.ui.adapters

import com.jakebarnby.filemanager.models.SourceFile
import com.jakebarnby.filemanager.util.TreeNode

@FunctionalInterface
interface OnSearchResultClicked {
    fun navigateToFile(fileNode: TreeNode<SourceFile>)
}